.github/workflows/cpp_clang_tidy.yml

```diff
@@ -0,0 +1,104 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Clang Tidy Check
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/cpp_clang_tidy.yml'
+      - 'cpp/**'
+
+env:
+  ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+  MVN_CMD: 'mvn -ntp'
+  WGET_CMD: 'wget -nv'
+  SETUP: 'source .github/workflows/util/setup-helper.sh'
+  CCACHE_DIR: "${{ github.workspace }}/.ccache"
+  # spark.sql.ansi.enabled defaults to false.
+  SPARK_ANSI_SQL_MODE: false
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build-native-lib-centos-7:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Ccache
+        uses: actions/cache/restore@v4
+        with:
+          path: '${{ env.CCACHE_DIR }}'
+          key: ccache-centos7-release-default-${{github.sha}}
+          restore-keys: |
+            ccache-centos7-release-default
+      - name: Build Gluten native libraries
+        run: |
+          docker pull apache/gluten:vcpkg-centos-7
+          docker run -v $GITHUB_WORKSPACE:/work -w /work apache/gluten:vcpkg-centos-7 bash -c "
+            set -e
+            yum install tzdata -y
+            df -a
+            cd /work
+            export CCACHE_DIR=/work/.ccache
+            mkdir -p /work/.ccache
+            bash dev/ci-velox-buildstatic-centos-7.sh
```

Comment:
It seems redundant to run the build for native code. Maybe, we can use the following action to reuse the binary produced in the main workflow. Please verify it.

```
- name: Download artifact from in-progress workflow
  uses: dawidd6/action-download-artifact@v3
  with:
    github_token: ${{secrets.GITHUB_TOKEN}}
    workflow: velox_backend_x86.yml
    name: velox-native-lib-centos-7-${{ github.sha }}
    commit: ${{ github.sha }}
    workflow_conclusion: ""
    check_artifacts: true
    search_artifacts: true
    if_no_artifact_found: warn
```

Reply:
Good ideal.

Reply:
If `dawidd6/action-download-artifact` requires that the base workflow must complete, I think it is acceptable, since our goal is to apply the clang tidy check and the time by clang tidy may not be too much if only the files changed by PR are checked.

Also note an existing workflow is using this action. 
https://github.com/apache/incubator-gluten/blob/4feac9bba366e032c7ad50fccc603d6fe3c29f8f/.github/workflows/test_report.yml#L34

Reply:
I found another problem: https://github.com/apache/incubator-gluten/actions/runs/20520437287

The action tj-actions/changed-files@v45 is not allowed in apache/incubator-gluten because all actions must be from a repository owned by your enterprise

I will use another method to implement this action.

Reply:
@xinghuayu007, you can choose another version verified by GitHub Marketplace, e.g., v46.

https://github.com/marketplace/actions/changed-files

Reply:
ok